File: 29-04/potencia.hs
potencia a b
    | b == 0 = 1
    | b > 0 = a * (potencia a (b - 1))

main :: IO () 
main = do
    putStrLn "Digite a base:"
    base <- readLn
    putStrLn "Digite o expoente:"
    expoente <- readLn
    putStrLn "A potencia é:"
    print (potencia base expoente)

File: 29-04/potencia.py
def potencia(a,b):
    if b == 0:
        return 1
    elif b > 0:
        return a * potencia(a, b-1)

if __name__ == "__main__":
    print("Digite primeiro a base e depois o expoente:")
    print("A potência é:",potencia(int(input()),int(input())))